### PR TITLE
Comments unit test

### PR DESCRIPTION
**Pull request by hechth**
Started implementing unit tests for files.
* Started comment feature unit test.

### CODE REVIEW

File: UnitTests/Schemas/IFC4X1/Comments/Data/csg-primitive-commented.ifc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+#1= IFCPROJECT('0xScRe4drECQ4DMSqUjd6d',#2,'proxy with CSG',$,$,$,$,(#3),#4);
+#2= IFCOWNERHISTORY(#6,#7,$,.ADDED.,1320688800,$,$,1320688800);
+#3= IFCGEOMETRICREPRESENTATIONCONTEXT($,'Model',3,1.0E-05,#8,$);
+#4= IFCUNITASSIGNMENT((#10,#11));
+#6= IFCPERSONANDORGANIZATION(#12,#13,$);
+#7= IFCAPPLICATION(#13,'1.0','IFC text editor','ifcTE');
+#8= IFCAXIS2PLACEMENT3D(#14,$,$);
+#9= IFCGEOMETRICREPRESENTATIONSUBCONTEXT('Body','Model',0,$,$,$,#3,$,.MODEL_VIEW.,$);
+#10= IFCSIUNIT($,.LENGTHUNIT.,.MILLI.,.METRE.);
```
Comment (pjanck) [inline UnitTests/Schemas/IFC4X1/Comments/Data/csg-primitive-commented.ifc]: This is wrong in the official example files.
```suggestion
#10= IFCSIUNIT(*,.LENGTHUNIT.,.MILLI.,.METRE.);

File: UnitTests/Schemas/IFC4X1/Comments/Data/csg-primitive-commented.ifc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+#14= IFCCARTESIANPOINT((0.,0.,0.));
+#15= IFCSHAPEREPRESENTATION(#9,'Body','CSG',(#18));
+#16= IFCDIMENSIONALEXPONENTS(0,0,0,0,0,0,0);
+#17= IFCMEASUREWITHUNIT(IFCPLANEANGLEMEASURE(0.017453293),#20);
+#18= IFCCSGSOLID(#21);
+#19= IFCPRODUCTDEFINITIONSHAPE($,$,(#15));
+
+// This time this is correct!
+
+#20= IFCSIUNIT($,.PLANEANGLEUNIT.,$,.RADIAN.);
```
Comment (pjanck) [inline UnitTests/Schemas/IFC4X1/Comments/Data/csg-primitive-commented.ifc]: See above.

File: UnitTests/Schemas/IFC4X1/Comments/Data/csg-primitive-commented.ifc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#21= IFCBLOCK(#23,1000.,1000.,2000.);
+#23= IFCAXIS2PLACEMENT3D(#24,$,$);
+#24= IFCCARTESIANPOINT((-500.,-500.,0.));
+
+/*
+Some other stuff that is described by a comment!
+*/
```
Comment (pjanck) [inline UnitTests/Schemas/IFC4X1/Comments/Data/csg-primitive-commented.ifc:38]: I would love to see an inline comment within an entity to complete the test. 
Sth like #77 = ENTITY (... /* comment */ , content,... );

File: UnitTests/Schemas/IFC4X1/Comments/Data/csg-primitive-commented.ifc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+#12= IFCPERSON($,'Liebich','Thomas',$,$,$,$,$);
+#13= IFCORGANIZATION($,'buildingSMART International',$,$,$);
+#14= IFCCARTESIANPOINT((0.,0.,0.));
+#15= IFCSHAPEREPRESENTATION(#9,'Body','CSG',(#18));
+#16= IFCDIMENSIONALEXPONENTS(0,0,0,0,0,0,0);
+#17= IFCMEASUREWITHUNIT(IFCPLANEANGLEMEASURE(0.017453293),#20);
+#18= IFCCSGSOLID(#21);
+#19= IFCPRODUCTDEFINITIONSHAPE($,$,(#15));
+
+// This time this is correct!
```
Comment (pjanck) [inline UnitTests/Schemas/IFC4X1/Comments/Data/csg-primitive-commented.ifc]: Add a comment at the end of some line with content as well.
Sth like #77 = ENTITY( content ); // This is a comment

**Review by pjanck — CHANGES_REQUESTED** [review]
Small changes to the example file requested in order to cover two additional possibilities of comments within an IFC file.

**Comment by pjanck** [comment]
Goes with #30 .

**Comment by hechth** [comment]
Fixed bugs in the file, test case succeeds now.

**Comment by hechth** [comment]
More work will be done in the future following #31

**Review by pjanck — APPROVED** [review]

**Comment by pjanck** [comment]
Resolves #20 .

**Pull request merged**